### DISCUSSION TITLE
pretixSCAN iOS - Offline not working

### DISCUSSION THREAD

**Discussion by vb-03** [pretix/pretix#3044] [Q&A]
Creating a discussion here as I'm not sure if there's something missing in my server/router config or if it's really a bug.
Basically, the iOS app only works online, isn't syncing anything. Android seems to work sometimes.
It shows the syncing, but the second I switch it to offline mode/loses connections, it throws an "Unknown ticket" error, and statistics are shown as zero.

However, the scanned ticket page shows right away the scan error:
![image](https://user-images.githubusercontent.com/47537857/213701615-11e8e0a8-9d36-4f49-b81c-a21274de9e68.png)
The logs of the device in question: (looks the same on every iOS device)
![image](https://user-images.githubusercontent.com/47537857/213702611-c081bbf9-519f-464e-ac79-d563d58dccaf.png)

**Comment by raphaelm** [comment]
"Test mode" orders are generally unavailable in offline mode, since test mode orders can "vanish" and the sync mechanism does not account for that

**Reply by vb-03** [reply]
After leaving test mode, it has indeed solved the Android issues.
However, iOS is still having issues. "Download orders" is turned on.
It already happened to me in another event (where we imported stuff into Pretix from another platform due to the offline scanning feature. We didn't end up using pretix that much there due to the QRCodes secrets from the other platform).
Online works perfectly, offline keeps throwing "invalid tickets". I put it online, let it download and then I either turn off the network or select the "Offline" mode in settings, I always get the same error.

**Reply by raphaelm** [reply]
Does the search function return any results when offline and entering e.g. a full order code?

**Reply by vb-03** [reply]
If I only turn the mobile data off, both search and statistics throws me "A data connection is currently not allowed."
When in offline mode, statistics are all at 0 (but the products names show up), and search returns nothing. 
Offline mode and mobile data on gives the same results.

**Reply by raphaelm** [reply]
Can you check your server logs (e.g. nginx access log) what URLs are hit by the app exactly when you run a forced full sync?

**Comment by vb-03** [comment]
iOS Logs:
![image](https://user-images.githubusercontent.com/47537857/214155037-e3899a49-7845-44ef-8fae-de3139839407.png)
Android Logs for comparison:
![image](https://user-images.githubusercontent.com/47537857/214155347-184341b5-65e2-43f6-aa09-9bf146a2d605.png)

(the red line separates app opening and forced full sync)

**Reply by raphaelm** [reply]
Hmm, it's weird that the iOS does the requests twice, but that shouldn't be the issue. Looking at the response size, it does download actual orders, so I'm currently out of ideas here what is going wrong there.

**Reply by vb-03** [reply]
Is there any other logs I can give you to help debug the issue?

**Reply by raphaelm** [reply]
Hmm, no, not sure if it's possible to get the app's internal logs from the app-store-installed version, I'm no iOS expert. I'll try to reproduce it when I'm working closer to the iOS app next time.

**Reply by vb-03** [reply]
If there's a non-app-store .IPA, I can try to sideload it with AltStore.

**Reply by raphaelm** [reply]
I just tried to reproduce this issue, but was unable to, offline mode works fine for me with the iOS app :(

**Reply by vb-03** [reply]
Oh, thanks for trying it out :(
We've been using mainly Android devices and were able to fix network issues on the server to use iOS devices, however I'm still intrigued to why it isn't working, can't stop thinking it's a firewall issue somehow...
